jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -109,3 +127,98 @@ pointer_typet select_pointer_typet::specialize_generics(
   }
   return pointer_type;
 }
+
+/// Return the first concrete type instantiation if any such exists. This method
+/// is only to be called when the `specialize_generics` cannot find an
+/// instantiation due to a loop in its recursion.
+/// \param parameter_name The name of the generic parameter type we want to have
+///   instantiated
+/// \param generic_specialization_map Map of type names to specialization stack
+/// \return The first instantiated type for the generic type or nothing if no
+/// such instantiation exists.
+optionalt<pointer_typet>
+select_pointer_typet::get_recursively_instantiated_type(
+  const irep_idt &parameter_name,
+  const generic_parameter_specialization_mapt
+    &generic_parameter_specialization_map) const
+{
+  generic_parameter_recursion_trackingt visited;
+  size_t depth = 0;
+  const size_t max =
+    generic_parameter_specialization_map.find(parameter_name)->second.size();
+
+  irep_idt current_parameter = parameter_name;
```

Comment:
if we make `irep_idt` to use string, it would be more efficient to use a reference